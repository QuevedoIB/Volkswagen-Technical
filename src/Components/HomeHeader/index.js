import React, { useCallback } from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { FaHeart, FaRegHeart } from "react-icons/fa";

import * as actions from "Redux/actions/cars";

import HomeTitle from "Components/HomeTitle";

const StyledHeaderContainer = styled.div`
  max-width: 100%;
  max-height: 40px;
  display: flex;
  justify-content: space-between;
  margin: ${({ theme }) => theme.defaultMargin}px 0;
`;

const StyledInput = styled.input`
  ${({ theme }) => {
    return `color: ${theme.primaryBlue};
    border: 1px solid ${theme.primaryBlue};
    font-family: ${theme.primaryFont};
    `;
  }}
  height: 40px;
  padding: 0px 20px;
`;

const StyledButton = styled.button`
  ${({ theme }) => {
    return `color: ${theme.primaryBlue};
    border: 1px solid ${theme.primaryBlue};
    font-family: ${theme.primaryFont};
    `;
  }}
  background-color: white;
  cursor: pointer;
  height: 42px;
  width: 100px;
`;

const StyledButtonContentContainer = styled.span`
  width: 100%;
  display: flex;
  align-items: flex-end;
  justify-content: space-evenly;
`;

const HomeHeader = ({ editFilters, filters: { keyword, liked }, amount }) => {
  const onChangeKeyword = useCallback(
    ({ target: { value } }) => {
      editFilters({
        keyword: value,
        liked,
      });
    },
    [editFilters, liked]
  );
  const onFavoritesPress = useCallback(
    () => editFilters({ keyword, liked: !liked }),
    [editFilters, keyword, liked]
  );

  return (
    <>
      <HomeTitle amount={amount} />
      <StyledHeaderContainer>
        <StyledInput
          value={keyword}
          placeholder="Buscar Marca"
          onChange={onChangeKeyword}
        />
        <StyledButton>
          <StyledButtonContentContainer onClick={onFavoritesPress}>
            Favoritos
            {liked ? <FaHeart /> : <FaRegHeart />}
          </StyledButtonContentContainer>
        </StyledButton>
      </StyledHeaderContainer>
    </>
  );
};

const mapStateToProps = ({ cars: { filters } }) => ({
  filters,
});

const mapDispatchToProps = (dispatch) => ({
  ...bindActionCreators(actions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeHeader);
