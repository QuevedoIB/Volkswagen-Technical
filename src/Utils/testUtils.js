import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";

import theme from "Styles/theme";
import store from "Redux/store";

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };

export { store };
