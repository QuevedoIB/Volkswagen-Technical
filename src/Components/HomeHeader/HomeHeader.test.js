import React from "react";
import { screen } from "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";
import { Context as ResponsiveContext } from "react-responsive";

import { store, render, fireEvent } from "Utils/testUtils";

import HomeHeader from "./index";

const mockedCarsData = [
  {
    Id: "001",
    Brand: "VOLKSWAGEN",
    CO2: "118",
    CV: 184,
    Energy: "DIESEL",
    Likes: 6,
    Kms: 39853,
    Model: "Polo",
    Image:
      "https://media.volkswagen.com/vw-AW1-my2020/iris?COSY-EU-100-1713VV%25lXm0kYNazzSU9mu0fO7Az6yyJ1vTvsd2SMppEKhjTtKSK8CPk1MM8H2Lvyr0Q%25UOVggAa5hP4KYwqeQeOOnH%25yyJ8H3WDZ4HvCJii8meZGoo0gZrnZlO4O6bLKKFNMbTggAcEvvFHOu5ZxxHlZkPF%25uAnW07ZZU8pVfRD%25xIIrsJYUP01bCCtBvyrXAhLLyH0kYbAwh88WTU9j%25I4bBMdPAA2Zjeqz5qMZKMI7UbSK8C7g21bBzdDAONX3qjBrr9UOXs3OHqttOWEC55PH4w99deVCZZslwCiiI%25KrffQXTJ22UFF4nn4hwFuugk8V00zPgAFF3ZWSxxcv0aBBhV%25TTTklhSWWHnk3GGKu9pJJMXF8llvhFkEELsGjVVbgvyXXY6XGttOoGv55PdJw99d4X%25ZZsQt8iiIiuRffQfZr22UDMynn4Y5SuugkqI00zjj5FF3hiNxxcf32BBhgBiTTkwsrWWHrIvGGK0BqJJMH0Ullvtd4EEL5AHVVbSS%25XXYQ%257ttOGzP55PAnL99deDQZZsoQpiiIprSffQoIr22UjMYnn4hGquug5wL00zr7AFF3wSFxxc7FhBBhyPnTTko2JWWHuq0GGK6kaJJMFG1llvld0EEL4flVVbf5PXXYWq9ttOG3s55PfZw99dg8aZZsExJiiIJkLffQfKr22UtZvnn45fsuugz3q00z4KAFF3yMSxxcBwBBBhj%25%25TTkbTzWWHK7vGGKGHLJJM0GYllvzTYEELA4wVVbN9WXXYZz8ttOGav55P44a99dDFWZZsaekiiI0GyffQ84t22Uam6nn4nzOuugkBa00z14hFF3FHSxxcm3oBBhPBjTTkmRfWWHurLGGK46jJJMRI%25llvrSZEEL3eaVVbr5WXXYTDMttOkWv55P40o99dKXsZZsa8C&width=864",
    Plate: "1234ABC",
    RegistrationDate: "01-05-2015",
    Liked: false,
  },
  {
    Id: "002",
    Brand: "VW",
    CO2: "118",
    CV: 184,
    Energy: "DIESEL",
    Likes: 6,
    Kms: 39853,
    Model: "Polo",
    Image:
      "https://media.volkswagen.com/vw-AW1-my2020/iris?COSY-EU-100-1713VV%25lXm0kYNazzSU9mu0fO7Az6yyJ1vTvsd2SMppEKhjTtKSK8CPk1MM8H2Lvyr0Q%25UOVggAa5hP4KYwqeQeOOnH%25yyJ8H3WDZ4HvCJii8meZGoo0gZrnZlO4O6bLKKFNMbTggAcEvvFHOu5ZxxHlZkPF%25uAnW07ZZU8pVfRD%25xIIrsJYUP01bCCtBvyrXAhLLyH0kYbAwh88WTU9j%25I4bBMdPAA2Zjeqz5qMZKMI7UbSK8C7g21bBzdDAONX3qjBrr9UOXs3OHqttOWEC55PH4w99deVCZZslwCiiI%25KrffQXTJ22UFF4nn4hwFuugk8V00zPgAFF3ZWSxxcv0aBBhV%25TTTklhSWWHnk3GGKu9pJJMXF8llvhFkEELsGjVVbgvyXXY6XGttOoGv55PdJw99d4X%25ZZsQt8iiIiuRffQfZr22UDMynn4Y5SuugkqI00zjj5FF3hiNxxcf32BBhgBiTTkwsrWWHrIvGGK0BqJJMH0Ullvtd4EEL5AHVVbSS%25XXYQ%257ttOGzP55PAnL99deDQZZsoQpiiIprSffQoIr22UjMYnn4hGquug5wL00zr7AFF3wSFxxc7FhBBhyPnTTko2JWWHuq0GGK6kaJJMFG1llvld0EEL4flVVbf5PXXYWq9ttOG3s55PfZw99dg8aZZsExJiiIJkLffQfKr22UtZvnn45fsuugz3q00z4KAFF3yMSxxcBwBBBhj%25%25TTkbTzWWHK7vGGKGHLJJM0GYllvzTYEELA4wVVbN9WXXYZz8ttOGav55P44a99dDFWZZsaekiiI0GyffQ84t22Uam6nn4nzOuugkBa00z14hFF3FHSxxcm3oBBhPBjTTkmRfWWHurLGGK46jJJMRI%25llvrSZEEL3eaVVbr5WXXYTDMttOkWv55P40o99dKXsZZsa8C&width=864",
    Plate: "1234ABC",
    RegistrationDate: "01-05-2015",
    Liked: true,
  },
  {
    Id: "003",
    Brand: "VW",
    CO2: "118",
    CV: 184,
    Energy: "DIESEL",
    Likes: 6,
    Kms: 39853,
    Model: "Polo",
    Image:
      "https://media.volkswagen.com/vw-AW1-my2020/iris?COSY-EU-100-1713VV%25lXm0kYNazzSU9mu0fO7Az6yyJ1vTvsd2SMppEKhjTtKSK8CPk1MM8H2Lvyr0Q%25UOVggAa5hP4KYwqeQeOOnH%25yyJ8H3WDZ4HvCJii8meZGoo0gZrnZlO4O6bLKKFNMbTggAcEvvFHOu5ZxxHlZkPF%25uAnW07ZZU8pVfRD%25xIIrsJYUP01bCCtBvyrXAhLLyH0kYbAwh88WTU9j%25I4bBMdPAA2Zjeqz5qMZKMI7UbSK8C7g21bBzdDAONX3qjBrr9UOXs3OHqttOWEC55PH4w99deVCZZslwCiiI%25KrffQXTJ22UFF4nn4hwFuugk8V00zPgAFF3ZWSxxcv0aBBhV%25TTTklhSWWHnk3GGKu9pJJMXF8llvhFkEELsGjVVbgvyXXY6XGttOoGv55PdJw99d4X%25ZZsQt8iiIiuRffQfZr22UDMynn4Y5SuugkqI00zjj5FF3hiNxxcf32BBhgBiTTkwsrWWHrIvGGK0BqJJMH0Ullvtd4EEL5AHVVbSS%25XXYQ%257ttOGzP55PAnL99deDQZZsoQpiiIprSffQoIr22UjMYnn4hGquug5wL00zr7AFF3wSFxxc7FhBBhyPnTTko2JWWHuq0GGK6kaJJMFG1llvld0EEL4flVVbf5PXXYWq9ttOG3s55PfZw99dg8aZZsExJiiIJkLffQfKr22UtZvnn45fsuugz3q00z4KAFF3yMSxxcBwBBBhj%25%25TTkbTzWWHK7vGGKGHLJJM0GYllvzTYEELA4wVVbN9WXXYZz8ttOGav55P44a99dDFWZZsaekiiI0GyffQ84t22Uam6nn4nzOuugkBa00z14hFF3FHSxxcm3oBBhPBjTTkmRfWWHurLGGK46jJJMRI%25llvrSZEEL3eaVVbr5WXXYTDMttOkWv55P40o99dKXsZZsa8C&width=864",
    Plate: "1234ABC",
    RegistrationDate: "01-05-2015",
    Liked: true,
  },
];

test("Home header renders correctly", () => {
  const container = render(<HomeHeader amount={0} />);
  expect(container.firstChild).toMatchSnapshot();
});

test("Searchbar text changes", () => {
  const testText = "Test";
  render(<HomeHeader amount={0} />);

  const input = screen.getByPlaceholderText("Buscar Marca");

  expect(input).toBeVisible();

  fireEvent.change(input, {
    target: { value: testText },
  });

  expect(input.value).toBe(testText);
});

test("Favorites changes icon correctly", async () => {
  render(<HomeHeader amount={0} />);

  expect(screen.getByTestId("disliked-icon")).toBeVisible();
  fireEvent.click(screen.getByTestId("like-button"));

  const {
    cars: { filters },
  } = store.getState();

  expect(filters.liked).toBeTruthy();
  expect(screen.getByTestId("liked-icon")).toBeVisible();
});

test("Removes big texts on mobile devices", async () => {
  const mobileSize = 599;
  render(
    <ResponsiveContext.Provider value={{ width: mobileSize }}>
      <HomeHeader amount={0} />
    </ResponsiveContext.Provider>
  );

  const favoritesText = screen.queryByText("Favorites");

  expect(favoritesText).not.toBeInTheDocument();
});
