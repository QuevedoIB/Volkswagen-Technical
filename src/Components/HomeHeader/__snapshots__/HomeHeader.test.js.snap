// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home header renders correctly 1`] = `undefined`;
